include/cudaq/Optimizer/CodeGen/Peephole.td

```diff
@@ -33,6 +33,23 @@ def XCtrlOneTargetToCNot : Pat<
     (LLVM_CallOp $callee, $args, $_, $_), (CreateCallCnot $args),
     [(InvokeOnXWithOneControl $callee, $args)]>;
 
+def InvokeOnZWithOneControl : Constraint<CPred<
```

Comment:
Please write this as a C++ class inheriting from a rewriter pattern. MLIR's support for autogenerating these has been kneecapped, and we need to rewrite all of these peepholes.

Reply:
> Please write this as a C++ class inheriting from a rewriter pattern. MLIR's support for autogenerating these has been kneecapped, and we need to rewrite all of these peepholes.
> 
> #476

Updated in 324588fa6.